Sources/YSwift/YText.swift

```diff
@@ -19,8 +19,12 @@ public final class YText: Transactable, YCollection {
     ///   - text: The string to append.
     ///   - transaction: An optional transaction to use when appending the string.
     public func append(_ text: String, in transaction: YrsTransaction? = nil) {
-        withTransaction(transaction) { txn in
-            self._text.append(tx: txn, text: text)
+        if let transaction {
+            self._text.append(tx: transaction, text: text)
+        } else {
+            withTransaction(transaction) { txn in
```

Comment:
Based on my experiences with ywasm - it would be better to have a single helper function, instead of if/else transaction check on every method. The number of operations in yrs API is huge and this kind of manual boilerplate on every single one of them will end with a royal PITA.

Reply:
I agree - but I wanted to get it working immediately for you so that you're unblocked. I'm not sure what @nugmanoff had planned when he set up this API